src/controllers/Auth.js

```diff
@@ -49,6 +52,72 @@ class Auth {
       return res.status(response.code).json(response);
     }
   }
+
+  /**
+   * @static
+   * @description Sends password reset mail
+   * @param {object} req Request Object
+   * @param {object} res Response Object
+   * @returns {object} JSON Response
+   */
+  static async forgotPassword(req, res) {
+    const { email } = req.body;
+    try {
+      const user = await User.findOne({ where: { email } });
+      if (!user) {
+        const response = new Response(false, 404, 'Email does not exist');
+        return res.status(response.code).json(response);
+      }
+      const { id, password: secret } = user.dataValues;
+      // create reset token
+      const resetToken = jwtHelper.generateToken({ userId: id }, secret, '1h');
+      // send mail
+      sendPasswordResetMail(req, user, resetToken);
+      const response = new Response(true, 200, 'Password reset mail sent', { resetToken });
+      return res.status(response.code).json(response);
+    } catch (error) {
+      const response = new Response(false, 500, 'Internal server error');
+      return res.status(response.code).json(response);
+    }
+  }
+
+  /**
+ * @static
+ * @description Reset's user password
+ * @param {object} req Request Object
+ * @param {object} res Response Object
+ * @returns {object} JSON Response
+ */
+  static async resetPassword(req, res) {
+    const { password } = req.body;
+    const { resetToken } = req.params;
+    // get user id from reset token;
+    const { payload } = jwt.decode(resetToken) || {};
+    const { userId } = payload || {};
+    // check if user exists
+    try {
+      const user = await User.findOne({ where: { id: userId || null } });
+      if (!user) {
+        const response = new Response(false, 404, 'User does not exist');
+        return res.status(response.code).json(response);
+      }
+      // get reset secret (use user's password as secret to make token one time use only)
+      const { password: secret } = user.dataValues;
+      jwt.verify(resetToken, secret);
```

Comment:
Seems you did not `await` the operation.

Reply:
jwt.verify() does not return a promise